### DISCUSSION TITLE
New command `tmt try` which will mimic current `1minutetip` menu

### DISCUSSION THREAD

**Discussion by lukaszachy** [teemtee/tmt#606] [Ideas]
Something can be done using aliases but I think it is too complicated compared to interactive menu.
I believe that this command can utilize existing steps, it will just make it more user friendly.
Otherwise one needs to be careful with 
- How to avoid 'finish' step
- Where to get IP of provisioned machine
- How to sync current dir (test) to the guest
- How to ignore all other tests and plans and keep just the one from PWD

Sure you can do everything with id, step selection but that is not easy to get right

**Comment by psss** [comment]
I like this. It could be used for both just reserving the guest for experimenting and debugging tests:

```
> tmt try

What would you like to do?

(p)rovision a guest, I want to play
(c)hoose provision method
(b)egin test debugging
(r)erun the test code
(e)nd the test session
...
> p
    provision
        how: minute
        image: fedora
        ...
```

**Reply by lukaszachy** [reply]
Hm, this could be much easier for new people. But we should make sure to not have too many options ;)

**Reply by lukaszachy** [reply]
I'll try to kick this off.

**Reply by psss** [reply]
> I'll try to kick this off.

Great, thanks!

> But we should make sure to not have too many options ;)

Agreed, so what about starting just with the default `virtual` provision and require the `image`?

    tmt try <distro>

Then we could directly decide:

* If `tmt test ls .` returns at least one test, execute it directly
* Otherwise just reserve a box for experimenting and log into it

Plus having `test`, `login` and `quit` in the menu?

**Reply by lukaszachy** [reply]
Actually, I'd prefer to focus on current, (internal) users with huge 1minutetip experience. So defaulting to minute (if installed) and slowly going  through current menu / cmdline options.

**Reply by psss** [reply]
> So defaulting to minute (if installed)

Understood, makes sense.

**Comment by psss** [comment]
This has been implemented by https://github.com/teemtee/tmt/pull/2443
